### DISCUSSION TITLE
__QUASAR_SSR_CLIENT__ is not defined

### DISCUSSION THREAD

**Discussion by r35p3ct** [quasarframework/quasar#17174] [CLI - SSR mode] [answered]
Hello. I have a small project on Quasar, apparently after updating one of the packages in package.json I am getting an error (__QUASAR_SSR_CLIENT__ is not defined). I do not use SSR.
How can this be fixed?

`{
  "name": "myapp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "dev": "vite  --host",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@quasar/extras": "^1.16.11",
    "axios": "^1.6.8",
    "jwt-decode": "^4.0.0",
    "moment": "^2.30.1",
    "pdfmake": "^0.2.10",
    "pinia": "^2.1.7",
    "quasar": "^2.16.1",
    "vue": "^3.4.27",
    "vue-axios": "^3.5.2",
    "vue-router": "^4.3.2",
    "xlsx": "https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz"
  },
  "devDependencies": {
    "@quasar/vite-plugin": "^1.7.0",
    "@vitejs/plugin-vue": "^5.0.4",
    "sass": "^1.77.0",
    "vite": "^5.2.11"
  }
}`

**Comment by tubescreamer808** [comment]
Hi,

Same problem for me when executing unit tests with vitest. I do not use SSR either


```
 FAIL  src/managers/AuthenticationManager.spec.ts [ src/managers/AuthenticationManager.spec.ts ]
ReferenceError: __QUASAR_SSR_CLIENT__ is not defined
 ❯ node_modules/quasar/dist/quasar.client.js:32:3
     30| // src/plugins/platform/Platform.js
     31| var isRuntimeSsrPreHydration = false ? { value: true } : ref(
     32|   __QUASAR_SSR_CLIENT__ && (__QUASAR_SSR_PWA__ ? document.body.getAttr…
       |   ^
     33| );
     34| var preHydrationBrowser;
 ❯ node_modules/@quasar/quasar-app-extension-testing-unit-vitest/src/helpers/install-quasar-plugin.ts:3:45

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

```
It appeared after doing `npm update`. 
My dependencies:
```
  "dependencies": {
    "axios": "^1.6.8",
    "pinia": "^2.1.7",
    "quasar": "^2.15.4",
    "quasar-app-extension-log-service": "file:../log-service",
    "vue": "^3.0.0",
    "vue-i18n": "^9.11.0"
  },
  "devDependencies": {
    "eventsourcemock": "^2.0.0",
    "@pinia/testing": "^0.1.3",
    "@quasar/app-vite": "^2.0.0-beta.2",
    "@quasar/quasar-app-extension-testing-unit-vitest": "^1.0.0",
    "@types/node": "^18.19.30",
    "@typescript-eslint/eslint-plugin": "^6.21.0",
    "@typescript-eslint/parser": "^6.21.0",
    "eslint": "^8.57.0",
    "eslint-config-prettier": "^8.10.0",
    "eslint-plugin-vue": "^9.24.1",
    "prettier": "^3.2.5",
    "typescript": "5.3.3",
    "vite-tsconfig-paths": "^4.3.2",
    "vitest": "^1.4.0",
    "vitest-mock-extended": "^1.3.1",
    "vue-tsc": "^2.0.12"
  }

```

**Reply by rstoenescu** [reply]
Add `@quasar/vite-plugin` to your vitest.config.js

**Reply by tubescreamer808** [reply]
It's already there as you can see below (unless you meant something else?). Those tests are in a Quasar app extension, can it be related to that?
Also, the tests are passing most of the time but that error happens "randomly" and on different test cases each time. 

```
import { defineConfig } from 'vitest/config';
import tsconfigPaths from 'vite-tsconfig-paths';
import vue from '@vitejs/plugin-vue';

import { quasar, transformAssetUrls } from '@quasar/vite-plugin';

export default defineConfig({
  test: {
    // set to true to enable watch mode (hot-reloading)
    watch: true,
    environment: 'happy-dom',
    setupFiles: 'test/vitest/setup-file.ts',
    include: [
      // Matches vitest tests in any subfolder of 'src' or into 'test/vitest/__tests__'
      // Matches all files with extension 'js', 'jsx', 'ts' and 'tsx'
      'src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
    ],
  plugins: [
    vue({
      template: { transformAssetUrls },
    }),
    quasar(),
    tsconfigPaths(),
  ],
});

```

**Reply by rstoenescu** [reply]
Can't work or fail only some of the times. It either works or fails spectacularly.
Regardless, I've pushed a commit in v2.16.2 which makes it work even when the q/vite-plugin is not installed. **BUT**, I would really suggest that you investigate thoroughly why it's not working for your setup with v2.16.1 first. It will uncover an error/misusage or something in your setup!

**Comment by hazwan91** [comment]
I have the same problem. I did the `npm update` and when i run vite using `npm run dev`, i am getting error `QUASAR_SSR_CLIENT`

package.json
```
"devDependencies": {
        "@inertiajs/vue3": "^1.0.0",
        "@quasar/extras": "^1.16.11",
        "@quasar/vite-plugin": "^1.7.0",
        "@ssthouse/vue3-tree-chart": "^0.2.6",
        "@tailwindcss/forms": "^0.5.3",
        "@vitejs/plugin-vue": "^5.0.0",
        "apexcharts": "^3.49.0",
        "autoprefixer": "^10.4.12",
        "axios": "^1.6.4",
        "laravel-vite-plugin": "^1.0",
        "postcss": "^8.4.31",
        "quasar": "^2.16.1",
        "sass": "^1.76.0",
        "tailwindcss": "^3.2.1",
        "v-viewer": "^3.0.13",
        "vite": "^5.0",
        "vue": "^3.4.0"
    }
```

EDIT:
For time being, to make my project work again, i need to change my package.json for both package like this
```
"@quasar/vite-plugin": "~1.6.0",
"quasar": "~2.15.4",
```

I did tried quasar version to 2.16.0 still have the same problem.
Assuming 2.16.0 above have something to do with this.

**Reply by rstoenescu** [reply]
You have installed @quasar/vite-plugin but you haven't added it to your vite.config.js

**Comment by rstoenescu** [comment, marked as answer]
@r35p3ct You have installed `@quasar/vite-plugin` but you haven't added it to your vite.config.js
